config_defaults_inc.php

```diff
@@ -2038,23 +2038,42 @@
 
 
 /**
- * Specifies the LDAP or Active Directory server to connect to.
+ * Specifies the LDAP or Active Directory server(s) to connect to.
+ * Multiple servers can be specified as a space-separated list.
  *
  * This must be a full LDAP URI (ldap[s]://hostname:port)
- * - Protocol can be either ldap or ldaps (for SSL encryption). If omitted,
- *   then an unencrypted connection will be established on port 389.
+ * - Protocol must be either: ldap (unencrypted or opportunistic StartTLS) or
+ *   ldaps (for TLS encryption).
  * - Port number is optional, and defaults to 389. If this doesn't work, try
  *   using one of the following standard port numbers: 636 (ldaps); for Active
  *   Directory Global Catalog forest-wide search, use 3268 (ldap) or 3269 (ldaps)
  *
  * Examples of valid URI:
- *   ldap.example.com
```

Comment:
If this is no longer valid, we need to consider what happens to all installations who are currently using this format - how will the system behave ?

Reply:
I removed it only because the docs here https://www.php.net/manual/en/function.ldap-connect.php say "Note that hostname:port is not a supported LDAP URI as the schema is missing."

Reply:
Ah, I see what you mean.  In fact, we're relying on the alternate, deprecated signature
*ldap_connect ( string $host = null , int $port = 389 ) : resource|false*

So technically it still works, but we don't know for how long... I'm not sure whether PHP throws any warnings when this is used, so I guess we should inform admins that this is usage is deprecated. 

This could be done in the ldap_connect_bind() function and/or as an Admin check.

Reply:
This seems like a confusing PHP thing to me. Both signatures take string as first argument, and the 2nd param has a default value of 389, so does calling `ldap_connect('some string')` call the 1st or 2nd variant? Seem ambiguous.

Anyway, I could log a message from `ldap_connect_bind` if the string doesn't start with `ldap[s]://`...

Reply:
> This seems like a confusing PHP thing to me.

Unfortunately, PHP is full of such inconsistencies. That may be one of the reasons why this usage is deprecated actually.

> log a message from `ldap_connect_bind` if the string doesn't start with `ldap[s]://`

Sounds good.

Reply:
Actually, `ldap_connect_bind` already checks that the string is a URI, and we already log if that fails.  So I've added an Admin Check instead.  (It's impressive what can be done by grepping and copy-pasting existing code :) )

Reply:
Indeed :wink: